controller/controller.go

```diff
@@ -124,6 +125,9 @@ var unitExp = regexp.MustCompile("@.service")
 // 	 bar@2.service
 //
 func (r Request) ExtendSlices() (Request, error) {
+	if len(r.SliceIDs) == 0 {
+		return r, nil
+	}
```

Comment:
the `ExtendedSlices` function iterates over `SliceIDs`. If that is nil or zero `Units` in the returned struct would be empty. Thus no files would be submitted to fleet. 

This change ensures the returned struct has the correct Units set.